app.py

```diff
@@ -246,25 +260,27 @@ def discard_inspection(inspection_id):   # pragma: no cover
 @auth.login_required
 @cross_origin(origins=FRONTEND_URL)
 @swag_from("docs/swagger/search_inspection.yaml")
-def search_inspections():   # pragma: no cover
-    return jsonify(error="Not yet implemented!"), HTTPStatus.SERVICE_UNAVAILABLE
-    # try:
-    #     # Database cursor
-    #     cursor = CONN.cursor()
-
-    #     # The search query used to find the label.
-    #     user_id = request.args.get('user_id')
-    #     label_id = request.args.get('label_id')
-    #     query = SearchQuery(user_id=user_id, label_id=label_id)
-
-    #     # TO-DO Send that search query to the datastore
-    #     inspections = inspection.get_all_user_inspection(cursor, query.user_id)
-    #     return jsonify(inspections), HTTPStatus.OK
-    # except Exception as err:
-    #     CONN.rollback()
-    # logger.error(f"Error occurred: {err}")
-    # logger.error("Traceback: " + traceback.format_exc())
-    #     return jsonify(error=str(err)), HTTPStatus.INTERNAL_SERVER_ERROR
+def search_inspection():   # pragma: no cover
+    try:
+        # Database cursor
+        with connect_db(FERTISCAN_DB_URL, FERTISCAN_SCHEMA) as conn:
+            with conn.cursor() as cursor:
+                # create_search_path(conn, cursor, FERTISCAN_SCHEMA)
+            
+                # The search query used to find the label.
+                username = auth.username()
+                # query = SearchQuery(username=username)
+
+                logger.info(f"Fetching user ID for username: {username}")
+                db_user = asyncio.run(get_user(cursor, username))
+
+                # TO-DO Send that search query to the datastore
+                inspections = fertiscan.inspection.get_all_user_inspection(cursor, db_user.id)
```

Comment:
You should not need to dig into the queries function to execute this. We made generic high level function to help out depending on the logic
```suggestion
                # verified = True/false depending if you want the finished or unfinished inspection
                inspections = fertiscan.get_user_analysis_by_verified(cursor, db_user.id, verified)
```